devicemodel/core/vmmapi.c

```diff
@@ -150,6 +226,11 @@ vm_create(const char *name, uint64_t req_buf, int *vcpu_num)
 		create_vm.vm_flag &= (~GUEST_FLAG_IO_COMPLETION_POLLING);
 	}
 
+	/* command line arguments specified CPU affinity overwrites HV's static configuration */
+	if (pcpu_affinity_bitmap != 0) {
+		create_vm.pcpu_bitmap = pcpu_affinity_bitmap;
+	}
```

Comment:
This is stranger.

BTW, please check API version somewhere.

Reply:
We need to check the API version.  No matter if pcpu_affinity_bitmap is 0 or not, we can assign it to create_vm.pcpu_bitmap, as if the version is right, though it may fail.

Reply:
I believe this code is not checking whether it may fail

Reply:
> I believe this code is not checking whether it may fail

Either libvirt or launch script calls acrn-dm with the --pcpu_list command line option to specify the CPU affinity. I think acrn-dm simply passes this info to hypervisor and checks whether the IOCTL goes through or not.

While hypervisor will audit the CPU affinity when it's handling this hypercall.

> We need to check the API version. No matter if pcpu_affinity_bitmap is 0 or not, we can assign it to create_vm.pcpu_bitmap, as if the version is right, though it may fail.

acrn-dm doesn't issue get_platform_info hypercall, and libvirt does. I think libvirt can check the HC version.  Since the --pcpu_list command could come from libvirt, or user manually adds to launch script, checking HC version here can't help too much I think.

Yes, "if (pcpu_affinity_bitmap != 0)" can be removed.